bundles/org.openhab.core/src/main/java/org/openhab/core/internal/i18n/I18nProviderImpl.java

```diff
@@ -344,39 +377,53 @@ public SystemOfUnits getMeasurementSystem() {
-
-        Map<SystemOfUnits, Unit<? extends Quantity<?>>> pressureMap = new HashMap<>();
-        pressureMap.put(SIUnits.getInstance(), HECTO(SIUnits.PASCAL));
-        pressureMap.put(ImperialUnits.getInstance(), ImperialUnits.INCH_OF_MERCURY);
-        dimensionMap.put(Pressure.class, pressureMap);
-
-        Map<SystemOfUnits, Unit<? extends Quantity<?>>> speedMap = new HashMap<>();
-        speedMap.put(SIUnits.getInstance(), SIUnits.KILOMETRE_PER_HOUR);
-        speedMap.put(ImperialUnits.getInstance(), ImperialUnits.MILES_PER_HOUR);
-        dimensionMap.put(Speed.class, speedMap);
-
-        Map<SystemOfUnits, Unit<? extends Quantity<?>>> lengthMap = new HashMap<>();
-        lengthMap.put(SIUnits.getInstance(), SIUnits.METRE);
-        lengthMap.put(ImperialUnits.getInstance(), ImperialUnits.INCH);
-        dimensionMap.put(Length.class, lengthMap);
-
-        Map<SystemOfUnits, Unit<? extends Quantity<?>>> intensityMap = new HashMap<>();
-        intensityMap.put(SIUnits.getInstance(), Units.IRRADIANCE);
-        intensityMap.put(ImperialUnits.getInstance(), Units.IRRADIANCE);
-        dimensionMap.put(Intensity.class, intensityMap);
-
-        Map<SystemOfUnits, Unit<? extends Quantity<?>>> percentMap = new HashMap<>();
-        percentMap.put(SIUnits.getInstance(), Units.ONE);
-        percentMap.put(ImperialUnits.getInstance(), Units.ONE);
-        dimensionMap.put(Dimensionless.class, percentMap);
-
-        Map<SystemOfUnits, Unit<? extends Quantity<?>>> angleMap = new HashMap<>();
-        angleMap.put(SIUnits.getInstance(), Units.DEGREE_ANGLE);
-        angleMap.put(ImperialUnits.getInstance(), Units.DEGREE_ANGLE);
-        dimensionMap.put(Angle.class, angleMap);
+        addDefaultUnit(Acceleration.class, Units.METRE_PER_SQUARE_SECOND);
+        addDefaultUnit(AmountOfSubstance.class, Units.MOLE);
+        addDefaultUnit(Angle.class, Units.DEGREE_ANGLE, Units.DEGREE_ANGLE);
+        addDefaultUnit(Area.class, SIUnits.SQUARE_METRE, ImperialUnits.SQUARE_FOOT);
+        addDefaultUnit(ArealDensity.class, Units.DOBSON_UNIT);
+        addDefaultUnit(CatalyticActivity.class, Units.KATAL);
+        addDefaultUnit(DataAmount.class, Units.BYTE);
+        addDefaultUnit(DataTransferRate.class, Units.MEGABIT_PER_SECOND);
+        addDefaultUnit(Density.class, Units.KILOGRAM_PER_CUBICMETRE);
+        addDefaultUnit(Dimensionless.class, Units.ONE);
+        addDefaultUnit(ElectricCapacitance.class, Units.FARAD);
+        addDefaultUnit(ElectricCharge.class, Units.COULOMB);
+        addDefaultUnit(ElectricConductance.class, Units.SIEMENS);
+        addDefaultUnit(ElectricConductivity.class, Units.SIEMENS_PER_METRE);
+        addDefaultUnit(ElectricCurrent.class, Units.AMPERE);
+        addDefaultUnit(ElectricInductance.class, Units.HENRY);
+        addDefaultUnit(ElectricPotential.class, Units.VOLT);
+        addDefaultUnit(ElectricResistance.class, Units.OHM);
+        addDefaultUnit(Energy.class, Units.JOULE);
```

Comment:
Could it be this change causes all kind of channels that are `kWh` to display in `J`, which is not expected. Is there a fix possible?

Reply:
Could this mean `%unit%` should not be used in binding xml definitions anymore?

Reply:
Could be. However, there is no way to avoid that and it was always the case with the dimensions that had a default unit before.

The state description takes precedence over the default, so we could check if we need to adjust `%unit%`. I'll have a look.

Reply:
@J-N-K The difference to other dimensions is that this here now is a breaking change - many of my items now show `J` instead of `KWh` and even a couple of my rules are impacted. I assume @Hilbrand (and many others) have a similar situation.
I wonder whether we should maybe define `kWh` as a default for `Energy` - after all, this is the unit that is almost always used in a smart home context, while `J` is rather a very rare exception
Wdyt?

Reply:
@kaikreuzer for the channels with `%unit%` my issue was solved in #3201.

Reply:
While the `J` might be the standard unit I think in a home automation context `kWh`should absolutely be the default.
99% of the time users want to compare/measure energy consumption of devices and that's typically done in `kWh`